python/cuvs/cuvs/neighbors/cagra/cagra.pyx

```diff
@@ -170,10 +173,8 @@ cdef class IndexParams:
                  nn_descent_niter=20,
                  compression=None):
 
-        # todo (dgd): enable once other metrics are present
-        # and exposed in cuVS C API
-        # self.params.metric = _get_metric(metric)
-        # self.params.metric_arg = 0
+        self._metric = metric
+        self.params.metric = <cuvsDistanceType>DISTANCE_TYPES[metric]
```

Comment:
Can you check if the dictionary has an entry for "sqeuclidean" and not "euclidean"? I'm not sure how the tests were working until now.

Reply:
The tests until now were using the `L2Expanded` metric for CAGRA irrespective of what was being passed as the metric. "sqeuclidean" translates to `L2Expanded` and "euclidean" translates to `L2SqrtExpanded`. The min recall bound in the pytests so far was quite low (=0.7) which is why even if the groundtruth was created using inner product, the tests were passing. I've checked precisely what cuvs distance type is being dispatched to the C++ level.